app/src/main/java/org/schabi/newpipe/info_list/dialog/InfoItemDialog.java

```diff
@@ -252,10 +252,12 @@ public Builder setAction(@NonNull final StreamDialogDefaultEntry entry,
          * @return the current {@link Builder} instance
          */
         public Builder addEnqueueEntriesIfNeeded() {
-            if (PlayerHolder.getInstance().isPlayQueueReady()) {
+            final PlayerHolder holder = PlayerHolder.getInstance();
+            if (holder.isPlayQueueReady()) {
                 addEntry(StreamDialogDefaultEntry.ENQUEUE);
 
-                if (PlayerHolder.getInstance().getQueueSize() > 1) {
+                final int size = holder.getQueueSize();
+                if (size > 1 && holder.getQueuePosition() < size - 1) {
```

Comment:
I think you don't need the `size > 1` condition. If `size <= 1`, then `size - 1 <= 0`, and since `queuePosition` is always `>= 0`, it is never `< size - 1 <= 0`.
```suggestion
                if (holder.getQueuePosition() < holder.getQueueSize() - 1) {
```